src/User/User.php

```diff
@@ -747,4 +748,30 @@ public static function getNotificationPreferenceKey($type, $method)
     {
         return 'notify_'.$type.'_'.$method;
     }
+
+    /**
+     * Refresh the user's comments count.
+     *
+     * @return $this
+     */
+    public function refreshCommentsCount()
+    {
+        $this->comments_count = $this->posts()->count();
+        $this->save();
+
+        return $this;
+    }
+
+    /**
+     * Refresh the user's comments count.
+     *
+     * @return $this
+     */
+    public function refreshDiscussionsCount()
+    {
+        $this->discussions_count = Discussion::where('start_user_id', $this->id)->count();
```

Comment:
I'm unsure why we are using the relation `posts()` in `refreshCommentsCount` and are using a direct query here. I would personally prefer a new relation `discussionsStarted`, what do you think @flarum/core ?

If we do, it needs:

- [ ] the relation
- [ ] the phpdoc change

Reply:
Yep. Not sure why we don't already have it. It can just be called `discussions`.